Add editors via the work settings page

awead opened the pull request:
Adds a new editors form to the work settings page where users can grant groups and users edit access to their works.

Fixes #277

mtribone commented:
<img width="1420" alt="screen shot 2017-03-07 at 11 47 11 am" src="https://cloud.githubusercontent.com/assets/4163828/23669340/c7e58eba-0331-11e7-9f3a-dc7d5387afca.png">

mtribone commented:
<img width="1419" alt="screen shot 2017-03-07 at 11 42 14 am" src="https://cloud.githubusercontent.com/assets/4163828/23669355/d2b7c362-0331-11e7-99a2-b54b11582395.png">

awead commented on app/views/curation_concerns/base/_form_progress.html.erb:
```diff
@@ -5,11 +5,15 @@
         <ul class="requirements">
-          <li class="incomplete" id="required-metadata">Enter required metadata</li>
+          <li class="incomplete" id="required-metadata">
+            <a href="#metadata"><%= t("curation_concerns.base.form_progress.metadata_required") %></a>
```
@mtribone could there be a line somewhere in feature or view test that checks for the presence of this link? Just add something to an existing test.

awead commented on app/views/curation_concerns/base/_form_progress.html.erb:
```diff
@@ -5,11 +5,15 @@
           <% if Sufia.config.work_requires_files %>
-            <li class="incomplete" id="required-files">Add files</li>
+            <li class="incomplete" id="required-files">
+              <a href="#files"><%= t("curation_concerns.base.form_progress.files_required") %></a>
```
same here.

mtribone commented on spec/features/generic_work/upload_and_delete_spec.rb:
```diff
@@ -64,6 +64,8 @@
         expect(page).to have_no_css("#generic_work_contributor")
 
         within("#savewidget") do
+          expect(page).to have_selector("a[href='#metadata']")
```
Here they be @awead ↕️

awead committed:
Add editors via the work settings page

Adds a new editors form to the work settings page where users can grant
groups and users edit access to their works.

rschenk reviewed (approved):
Looks good to me. I'm glad the pattern I set up for the Work Settings page seems to be reasonable
